Authentication/SignUp/View/SignupControllerDelegate.swift

```diff
@@ -53,19 +53,13 @@ extension SignupControllerDelegate {
         app.networkActivityIndicatorVisible = false
     }
     
-    public func signupController(controller: SignupController, didSignUpWithError error: SessionServiceError) {
-        controller.signupView.usernameTextFieldValid = false
-        controller.signupView.emailTextFieldValid = false
-        controller.signupView.passwordTextFieldValid = false
-        controller.signupView.passwordConfirmationTextFieldValid = false
-        controller.signupView.signupButtonPressed = false
-        
-        if let errorLabel = controller.signupView.signupErrorLabel {
+    public func signupController(controller: SignupController, didSignUpWithError error: SessionServiceError) {        
+        if let errorLabel = controller.signupView.signUpErrorLabel {
             errorLabel.text = error.message
         }
         if shouldDisplaySignupErrorWithAlert {
```

Comment:
I wouldn't do that because there are many possible validation errors, but this refers to only sign up errors, and it's not representing displaying the error in any form, but with an alert. Because in the label it will always display it.